### DISCUSSION TITLE
How to include .cls, .sty file inside latax blade template

### DISCUSSION THREAD

**Discussion by nprasath002** [ismaelw/laratex#25] [Q&A] [answered]
Hi there,

First of all thanks for the amazing package. I am building an app that takes user input and generates pdf documents.
I am testing the app with this template https://github.com/latex-ninja/hipster-cv. I included the `main.tex` inside a blade template and placed `hipstercv.cls` and `hipstercv.sty` inside the same directory.

While trying to compile latex throws the following error
```
This is pdfTeX, Version 3.141592653-2.6-1.40.22 (TeX Live 2022/dev/Debian) (preloaded format=pdflatex) restricted \write18 enabled. entering extended mode (/var/www/html/storage/app/f9ZczJrQkMWrTiwa LaTeX2e <2021-11-15> patch level 1 L3 programming layer <2022-01-21> ! LaTeX Error: File `/Users/nprasath002/PhpstormProjects/miro/resources/views/f ilament/employee/cv-templates/simplehipstercv.cls' not found. Type X to quit or <RETURN> to proceed, or enter new name. (Default extension: cls) Enter file name: ! Emergency stop. <read *> l.6 .../employee/cv-templates/simplehipstercv.cls} ^^M ! ==> Fatal error occurred, no output PDF file produced! Transcript written on /var/www/html/storage/app//f9ZczJrQkMWrTiwa.log.
```

I tried giving the a[bosulte path](https://github.com/ismaelw/laratex#using-graphics-inside-of-your-latex-files) with no luck.

Here is the output of the log file
```
This is pdfTeX, Version 3.141592653-2.6-1.40.22 (TeX Live 2022/dev/Debian) (preloaded format=pdflatex 2023.10.6)  9 OCT 2023 08:14
entering extended mode
 restricted \write18 enabled.
 %&-line parsing enabled.
**/var/www/html/storage/app/f9ZczJrQkMWrTiwa
(/var/www/html/storage/app/f9ZczJrQkMWrTiwa
LaTeX2e <2021-11-15> patch level 1
L3 programming layer <2022-01-21>

! LaTeX Error: File `/Users/nprasath002/PhpstormProjects/miro/resources/views/f
ilament/employee/cv-templates/simplehipstercv.cls' not found.

Type X to quit or <RETURN> to proceed,
or enter new name. (Default extension: cls)

Enter file name: 
! Emergency stop.
<read *> 
         
l.6 .../employee/cv-templates/simplehipstercv.cls}
                                                  ^^M
End of file on the terminal!

 
Here is how much of TeX's memory you used:
 23 strings out of 478287
 889 string characters out of 5849289
 289230 words of memory out of 5000000
 18315 multiletter control sequences out of 15000+600000
 469259 words of font info for 28 fonts, out of 8000000 for 9000
 1141 hyphenation exceptions out of 8191
 16i,0n,26p,280b,18s stack positions out of 5000i,500n,10000p,200000b,80000s
!  ==> Fatal error occurred, no output PDF file produced!
```

Let me know any ideas on how to resolve this.

Thanks a lot

**Comment by ismaelw** [comment]
Hi @nprasath002 

Thank you very much for your message.
I never had a project where Classes or Styles where necessary. Let me try if I can make it work in my test environment.

As soon as I have something to share I will get back to you.

Kind regards,
Ismael

**Comment by ismaelw** [comment, marked as answer]
Hi @nprasath002 

Okay I am done testing for now.
In my environment I did the following:

- Create a latex blade file
- Create a class (that just defines a color)
- Create a style (that just loads a package)

This is my conclusion:
The path where your pdflatex executable runs in is inside the temp folder defined in the config.
In my laratex.php config file I defined the following temp Path: `'tempPath' => 'app/temp/'`

Therefore the classes and styles are beeing searched inside the storage/app/temp/ folder and not in the views folder.

This is my setup:
- /resources/views/latex/main.blade.php
- /storage/app/temp/testclass.cls
- /storage/app/temp/teststyle.sty

This is the source code on how I call the LaraTex rendering:
```
return (new LaraTeX('latex.main'))->with([
	'Name' => 'John Doe'
])->download('test_classes_styles.pdf');
```

This is the source code of my main.blade.php file:
```
\documentclass[testcolor]{testclass}

\begin{document}
	\textbf{Test Document for classes and styles}
	@latex($Name)
\end{document}
```

This is the source code of my testclass.cls file:
```
\NeedsTeXFormat{LaTeX2e}
\ProvidesClass{testclass}[2023/10/09 Test LaTeX class]

\LoadClass[]{article}

\RequirePackage{xcolor}

\DeclareOption{testcolor}{
	\definecolor{textColor}{HTML}{2E2E2D}
}

\ProcessOptions\relax

\usepackage{teststyle}
```

This is the source code of my teststyle.sty file:
`\usepackage[T1]{fontenc}`

 
I think you might need to move your classes and styles accordingly and from there go on with the troubleshooting. In my projects I only used single tex files without styles or classes (because all the configuration is built in the main.tex file in my case). For included graphics I used the absolute path which also always worked. 

I am not sure why using an absolute path to the views folder didn't worked in your case. That might have to do with your pdflatex executable and where it looks for files. My assumption is that pdflatex doesn't like to look outside of its working directory.

Can you try that and report back to me if it worked for you?

Kind regards,
Ismael

**Reply by nteej** [reply]
@ismaelw ,I also in need of this solution & get tried as you have advised above & it works like a charm.Simple solution with huge impact on the cleaness of the tex template for LaTeX pdf building.
I've tried on Mac environment based Latex Packages(LiveLatex) & centos & Debian too.I'm confirming its working as desired without complaints.

**Comment by nprasath002** [comment]
You are a hero @ismaelw . That worked. Would be awesome to have options to pass these as additional arguments.

Something like
```
return (new LaraTeX('latex.main'))->class('latex.testclass')->style('latext.teststyle')->with([
	'Name' => 'John Doe'
])
```

or 

```
return (new LaraTeX('latex.main' , [ 'class' => 'latex.testclass', 'style' => 'latext.teststyle' ]))->with([
	'Name' => 'John Doe'
])
```

Right now log files are outputted in the same directory as i place the style and class files which makes this messy

**Reply by ismaelw** [reply]
Hi @nprasath002  

No worries! Happy to help!

I believe this won't be possible. As far as I know the pdflatex compiler doesn't come with command parameters to pass cls and sty files. So you will always have to include them in the tex blade file and I do not see a way how the package could know where to insert the cls and sty class imports.

This package comes with a teardown feature. If this is enabled in the config the package removes all files like log etc. that are being created while rendering the PDF so the temp path is not so cluttered.

What you can also do is to create a folder inside of the temp folder where you store the cls and sty files and pass the path to that folder.
This is how I do it with my graphics.
